src/main/java/tutorlink/command/ListGradeCommand.java

```diff
@@ -1,17 +1,51 @@
 package tutorlink.command;
 
 import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.Comparator;
 
 import tutorlink.appstate.AppState;
 import tutorlink.result.CommandResult;
+import tutorlink.grade.Grade;
 
 public class ListGradeCommand extends Command {
 
     public static final String COMMAND_WORD = "list_grade";
+    private static final String MESSAGE_NO_GRADES = "No grades have been recorded yet.";
 
     @Override
     public CommandResult execute(AppState appState, HashMap<String, String> hashMap) {
-        return new CommandResult(appState.grades.toString());
+        ArrayList<Grade> grades = appState.grades.getGradeArrayList();
+
+        if (grades.isEmpty()) {
+            return new CommandResult(MESSAGE_NO_GRADES);
```

Comment:
Should we extract this to Commons?

Reply:
So I can change to print a line instead?

Reply:
No as in you can extract `MESSAGE_NO_GRADES` to Commons library, assuming that it will be used again. 

Otherwise it is fine to just leave it as is.